lib/spack/spack/ci.py

```diff
@@ -705,14 +722,51 @@ def generate_gitlab_ci_yaml(
             files (spack.yaml, spack.lock), etc should be written.  GitLab
             requires this to be within the project directory.
     """
+    rev1, rev2 = get_change_revisions()
+    tty.warn(f"Got following revisions: rev1={rev1}, rev2={rev2}")
+
+    # Get the joined "ci" config with all of the current scopes resolved
+    ci_config = cfg.get("ci")
+    spack_prune_untouched = os.environ.get("SPACK_PRUNE_UNTOUCHED", None)
+
+    changed = rev1 and rev2
+    affected_pkgs = None
+    if spack_prune_untouched and changed:
+        affected_pkgs = compute_affected_packages(rev1, rev2)
+        tty.debug("affected pkgs:")
+        for p in affected_pkgs:
+            tty.debug(f"  {p}")
+
+        possible_builds = spack.package_base.possible_dependencies(*env.user_specs)
+        # with env.read_transaction():
+        changed = any((spec in p for p in possible_builds.values()) for spec in affected_pkgs)
+
+        if not changed:
+            spack_ci = SpackCI(ci_config)
+            spack_ci_ir = spack_ci.generate_ir()
+
+            # No jobs should be generated.
+            noop_job = spack_ci_ir["jobs"]["noop"]["attributes"]
+            # If this job fails ignore the status and carry on
+            noop_job["retry"] = 0
+            noop_job["allow_failure"] = True
+
+            tty.debug("No specs to rebuild, generating no-op job")
```

Comment:
I think this print message should be distinct from the already-existing one below, so we can tell the difference between the two cases that can result in a no-op job.

Reply:
I think the "affected packages" debug before should be enough right? In the case that noop is generated otherwise we will see more output than that explaining what was checked and why.

Reply:
I don't understand how the "affected packages" debug output has anything to do with this.  You're right that the next round of untouched pruning would at least print "Affected specs:" if we got that far.  But why can't this just say "Skipping concretization, generating no-op job" or similar?

Reply:
I added more context to the debug for affected packages.